cluster-controller/src/main/java/io/strimzi/controller/cluster/operations/openshift/S2IOperation.java

```diff
@@ -1,19 +1,56 @@
 package io.strimzi.controller.cluster.operations.openshift;
 
+import io.fabric8.openshift.client.OpenShiftClient;
 import io.strimzi.controller.cluster.resources.Source2Image;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
 
 /**
  * Base Source2Image operation
  */
-public abstract class S2IOperation extends OpenShiftOperation {
-    protected final Source2Image s2i;
+public abstract class S2IOperation {
```

Comment:
The `S2IOperation` in its own package kind of stands out. I think we should either integrate it into the `KafkaConnectClusterOperation` or we should maybe try to change it into `ClusterOperation`. Right now I don't think we have any plans to add S2I support anywhere else, so maybe we can do the first.

Reply:
FYI: We can also do it later in separate PR.